src/NuGet.Services.SearchService/App_Start/WebApiConfig.cs

```diff
@@ -94,25 +94,30 @@ private static IDependencyResolver GetDependencyResolver(HttpConfiguration confi
             return new AutofacWebApiDependencyResolver(container);
         }
 
-        private static IConfigurationRoot GetConfigurationRoot(string configurationFilename, out ISecretInjector secretInjector)
+        private static IConfigurationRoot GetConfigurationRoot()
         {
-            var basePath = HostingEnvironment.MapPath("~/");
+            const string prefix = "APPSETTING_";
+            var jsonFile = Path.Combine(HostingEnvironment.MapPath("~/"), @"Settings\local.json");
 
-            var builder = new ConfigurationBuilder()
-                .SetBasePath(basePath)
-                .AddJsonFile(configurationFilename, optional: false, reloadOnChange: false);
-
-            var uninjectedConfiguration = builder.Build();
+            // Load the configuration without injection. This allows us to read KeyVault configuration.
+            var uninjectedBuilder = new ConfigurationBuilder()
+                .AddJsonFile(jsonFile) // The JSON file is useful for local development.
+                .AddEnvironmentVariables(prefix); // Environment variables take precedence.
```

Comment:
To clarify--does the last thing added take precedence?

Reply:
Yes.